riscv_isac/coverage.py

```diff
@@ -608,17 +654,22 @@ def compute_per_line(instr, cgf, xlen, addr_pairs,  sig_addrs):
                 rs1_hi_val = struct.unpack(unsgn_sz, bytes.fromhex(arch_state.x_rf[rs1+1]))[0]
                 rs1_val = (rs1_hi_val << 32) | rs1_val
         elif rs1_type == 'x':
-            rs1_val = struct.unpack(sgn_sz, bytes.fromhex(arch_state.x_rf[rs1]))[0]
-            if instr.instr_name in ["fmv.w.x","fcvt.s.l", "fcvt.s.lu"]:
-                rs1_val = '0x' + (arch_state.x_rf[rs1]).lower()
+            if instr.instr_name in ["fmv.w.x","fcvt.s.l","fcvt.s.lu","fcvt.d.w","fcvt.d.wu"]:
+                if arch_state.flen == 64:
+                    rs1val = int('0x' + (arch_state.x_rf[rs1]).lower(),16)
+                elif arch_state.flen == 32:
+                    rs1val = int('0x' + (arch_state.x_rf[rs1][-8:]).lower(),16)
+                rs1_val = twos_complement(rs1val,arch_state.flen) #To handle the signed integer values
+            else:
+                rs1_val = struct.unpack(sgn_sz, bytes.fromhex(arch_state.x_rf[rs1]))[0]
         elif rs1_type == 'f':
-            if instr.instr_name in ["fadd.s","fsub.s","fclass.s","fmul.s","fdiv.s","fsqrt.s","fmadd.s","fmsub.s","fnmadd.s","fnmsub.s","fmax.s","fmin.s","feq.s","flt.s","fle.s","fmv.x.w","fmv.w.x","fcvt.wu.s","fcvt.s.wu","fcvt.w.s","fcvt.s.w","fsgnj.s","fsgnjn.s","fsgnjx.s","fcvt.s.l", "fcvt.s.lu"]:
+            if arch_state.flen == 32: #This condition is extracting the last 8 characters of returned hexa value if the FLEN is 32 bits
```

Comment:
won't the else part get executed, if the condition is not met on the if block?

Reply:
`fadd.s` always works on single precision values. Hence the values have to be interpreted as a single precision number instead of double precision. `flen` denotes the maximum supported length on the implementation. So, in `RV64D` the `flen` will be `64`, but for `fadd.s` will need the register values to be interpreted as single precision.

Reply:
is it not correct to say that fadd.s is belongs to Single precision float only? So for double precision floating add, we have fadd.d which is in RV64D?

Reply:
`D` does not exist in isolation without `F`. So if the configuration is `RV64D`, all the `F` extension instructions are supported too. The implementation and by extension the logs will always hold `64` or `32` bit based on the maximum supported `flen`. The single precision ops are supported from the same `64` bit registers by NaN boxing.

Reply:
Current implementation of isac doesn't support this type of run, when we try to do fadd.s in RV64D. However, this would aid any future implementation what we do, so happy to make this up.